aws/rust-runtime/aws-inlineable/src/s3_express.rs

```diff
@@ -690,18 +690,18 @@ pub(crate) mod runtime_plugin {
                 .is_none()
             {
                 match env.get(env::S3_DISABLE_EXPRESS_SESSION_AUTH) {
-                    Ok(value)
-                        if value.eq_ignore_ascii_case("true")
-                            || value.eq_ignore_ascii_case("false") =>
-                    {
+                    Ok(value) if valid_environment_variable_value(&value) => {
                         let value = value
-                            .to_lowercase()
                             .parse::<bool>()
-                            .expect("just checked to be a bool-valued string");
+                            .expect("just checked to be a valid bool-valued string");
                         layer.store_or_unset(Some(crate::config::DisableS3ExpressSessionAuth(
                             value,
                         )));
                     }
+                    Ok(value) => panic!(
+                        "Must provide either `true` or `false` for an environment variable `{}`, but got {}",
+                        env::S3_DISABLE_EXPRESS_SESSION_AUTH,
+                        value),
```

Comment:
Should we log a warning and ignore it instead?

Reply:
Switched to ignore and warn f2c3445. Our intention is that any unexpected value for an env var shouldn't cause an app to crash.